### ISSUE TITLE
PlayerLoginEvent#allow is not working

### ISSUE THREAD

**Issue by RanMinecraft** [Dreeam-qwq/Leaf#85]
I cant allow player join server when server is full.
but work on paper latest version 1.21-90

```
This server is running Leaf version 1.21-DEV-ver/1.21@ed3cd2c (2024-07-18T03:00:31Z) (Implementing API version 1.21-R0.1-SNAPSHOT)
* You are 1 version(s) behind
Download the new version at: https://github.com/Winds-Studio/Leaf
Previous: 1.21-4-090775e (MC: 1.21)
```

```
@EventHandler(priority = EventPriority.HIGH)
    public void onPlayerLogin(PlayerLoginEvent event) {
        if (event.getResult() == PlayerLoginEvent.Result.KICK_FULL) {
            Player player = event.getPlayer();
            if (player.hasPermission("ranmc.joinfullserver")) {
                event.allow();
            }
        }
    }
```

**Comment by F3F5** [comment]
provide logs

**Comment by RanMinecraft** [comment]
> provide logs

```
UUID of player XYFCM is f05f62f1-533c-3326-bc41-739acf3b7f8c
[17:02:23 INFO]: com.mojang.authlib.GameProfile@719c51c[id=f05f62f1-533c-3326-bc41-739acf3b7f8c,name=XYFCM,properties={}] lost connection: The server is full!
[17:02:28 INFO]: UUID of player XYFCM is f05f62f1-533c-3326-bc41-739acf3b7f8c
[17:02:28 INFO]: com.mojang.authlib.GameProfile@2fe5d851[id=f05f62f1-533c-3326-bc41-739acf3b7f8c,name=XYFCM,properties={}] lost connection: The server is full!
```

**Comment by RanMinecraft** [comment]
Nothing special

> > provide logs
> 
> `UUID of player XYFCM is f05f62f1-533c-3326-bc41-739acf3b7f8c [17:02:23 INFO]: com.mojang.authlib.GameProfile@719c51c[id=f05f62f1-533c-3326-bc41-739acf3b7f8c,name=XYFCM,properties={}] lost connection: The server is full! [17:02:28 INFO]: UUID of player XYFCM is f05f62f1-533c-3326-bc41-739acf3b7f8c [17:02:28 INFO]: com.mojang.authlib.GameProfile@2fe5d851[id=f05f62f1-533c-3326-bc41-739acf3b7f8c,name=XYFCM,properties={}] lost connection: The server is full!`

**Comment by Dreeam-qwq** [comment]
You can give player `purpur.joinfullserver` permission to enable them join full server

**Comment by Dreeam-qwq** [comment]
Fixed in 0c0c4569120e5cf1e8d730b0b6c2ab1e539cc9ea
